### ISSUE TITLE
Apollo v3 merge function fails when spreading existing non-normalized data

### ISSUE THREAD

**Issue by 3nvi** [apollographql/apollo-client#6245]
In Apollo v3 we have the ability to specify a merge function in a particular type field. The merge function works fine when merging types that are normalized in the Cache, but throws an exception if (for whatever reason) you are attempting to spread the existing data.

For example, when defining a merge policy:

```ts
 merge: (existing, incoming) => {
    if (!existing) {
        return incoming;
    }

   // dummy one just for showcasing purposes
   return {
     ...existing,
   };
 },
```

Everything works fine if the type of the `existing` is something that's normalized in the InMemoryCache. If it's **not normalized** (by missing an `id`/ `_id` or by specifying `keyFields: false`), then I get the following error:

```js
policies.js:246 Uncaught (in promise) TypeError: Cannot assign to read only property '__typename' of object '[object Object]'
    at policies.js:246
    at Array.forEach (<anonymous>)
    at Policies.push../node_modules/@apollo/client/cache/inmemory/policies.js.Policies.applyMerges (policies.js:245)
    at policies.js:238
    at Array.map (<anonymous>)
    at Policies.push../node_modules/@apollo/client/cache/inmemory/policies.js.Policies.applyMerges (policies.js:238)
    at policies.js:246
    at Array.forEach (<anonymous>)
    at Policies.push../node_modules/@apollo/client/cache/inmemory/policies.js.Policies.applyMerges (policies.js:245)
    at policies.js:246
```

Funnily enough, if I attempt to spread the `incoming` instead of the `existing`, then everything seems fine. I'm including @benjamn since I believe this may relate to his work.

**Intended outcome:**
Merging of a field should be irrelevant of whether the data is normalized in the cache.

We should be allowed to do:

```ts
 merge: (existing, incoming) => {
    if (!existing) {
        return incoming;
    }

   return {
     ...incoming,
     items: [...existing.items, ...incoming.items]
   };
 },
```

without any issues.

**Actual outcome:**
If I attempt to write `...existing` in the result  in the merge function (i.e. `...existing`, `...existing.items`, etc.) then an error is triggered.

**How to reproduce the issue:**
It's pretty simple. Define a merge function that returns the example found in the description, for a type that is not normalized in the cache.

**Versions**

  System:
    OS: macOS 10.15.2
  Binaries:
    Node: 10.16.0 - /usr/local/bin/node
    Yarn: 1.19.0 - /usr/local/bin/yarn
    npm: 6.14.2 - /usr/local/bin/npm
  Browsers:
    Chrome: 81.0.4044.129
    Firefox: 72.0.2
    Safari: 13.0.4
  npmPackages:
    @apollo/client: 3.0.0-beta.41 => 3.0.0-beta.41
    apollo-link-error: ^1.1.12 => 1.1.12

**Comment by benjamn** [comment]
@3nvi Can you give `@apollo/client@3.0.0-beta.48` a try? My fix for this was also in `beta.47`, but `beta.48` is currently the latest version.

**Comment by 3nvi** [comment]
Hey @benjamn the current issue is indeed fixed on `@apollo/client@3.0.0-beta.48`, thanks a lot!

Another issue related to data mutations is still present though https://github.com/apollographql/apollo-client/issues/6202. I've also added a reproduction codesandbox using the latest beta.
